diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt

```diff
@@ -190,23 +190,46 @@ ${handleMultilineCopyright(copyrightText)}
         }
     }
 
+    /**
+     * Deletes all spaces and newlines
+     * Used to compare copyrights in yaml and file
+     */
+    private fun String.copyrightSense(): String =
+        replace("\n","")
+        .replace(" ","")
+
+    /**
+     * If it is multiline copyright, this method makes needed indents.
+     * Otherwise, if it is one line copyright, it returns it trimmed.
+     */
     private fun handleMultilineCopyright(copyrightText: String): String {
-        var properCopyright = ""
-        if (copyrightText.contains("\n")) {
-            copyrightText.split("\n").map { "    $it" }.forEach {
-                properCopyright += if (it == "    ") {
-                    "\n"
-                } else {
-                    "$it\n"
+        val firstLineIndent = copyrightText
+                .split("\n")
+                .let { list ->
+                    // points on the next not empty line
+                    val index = list.takeWhile { it.isEmpty() }.count()
+                    list[index].takeWhile { it == ' ' }
                 }
-            }
+        if (copyrightText.contains("\n")) {
+            return copyrightText
+                    .split("\n")
+                    .reduce { acc, nextLine ->
+                        when {
+                            nextLine.removePrefix(firstLineIndent).isEmpty() -> {
+                                "$acc\n"
+                            }
+                            // checks only first line
+                            acc.isEmpty() -> {
+                                nextLine.removePrefix(firstLineIndent)
+                            }
+                            else -> {
+                                "$acc\n    ${nextLine.removePrefix(firstLineIndent)}"
```

Comment:
I don't think it's correct to trim indent and hard-code indent of 4. Actually, yml supports custom indentation for multiline blocks in a form
```yml
node: |2
      value
```
In this case 2 leading spaces will be trimmed and other will be kept in the string after parsing. I think, copyright text can be written to the code as is, maybe only trimming blank lines.

For single-line `copyrightText` we can check if it start with spaces: if it does, than probably the user explicitly used the latter syntax. Otherwise, 4 spaces can be inserted.